src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/TransmissionStorage.cs

```diff
@@ -217,7 +242,7 @@ private IPlatformFile CreateTemporaryFile()
         private IEnumerable<IPlatformFile> GetTransmissionFiles()
         {
             IEnumerable<IPlatformFile> newFiles = this.folder.GetFiles();
-            newFiles = newFiles.Where(f => f.Extension == TransmissionFileExtension);
+            newFiles = newFiles.Where(f => f.Extension == TransmissionFileExtension && !this.badFiles.ContainsKey(f.Name));
```

Comment:
I've changed it to post filter only where the files are actually loaded into the queue so now the size calculation will include the known bad files so the folder can't grow indefinitely.